Problem with chrono version

```
    Updating registry `https://github.com/rust-lang/crates.io-index`
   Compiling num-traits v0.1.40
   Compiling void v1.0.2
   Compiling libc v0.2.29
   Compiling getopts v0.2.14
   Compiling semver v0.1.20
   Compiling byteorder v0.4.2
   Compiling bitflags v0.5.0
   Compiling cfg-if v0.1.2
   Compiling bitflags v0.4.0
   Compiling rand v0.3.16
   Compiling time v0.1.38
   Compiling num-integer v0.1.35
   Compiling pulldown-cmark v0.0.3
   Compiling rustc_version v0.1.7
   Compiling num-iter v0.1.34
   Compiling nix v0.6.0
   Compiling num v0.1.40
   Compiling chrono v0.4.0
   Compiling chrono v0.2.25
   Compiling tempdir v0.3.5
   Compiling skeptic v0.5.0
   Compiling i2cdev v0.3.1
   Compiling timer v0.1.6
   Compiling chipclock v0.1.0 (file:///home/xav/rust/chipclock)
error[E0308]: mismatched types
  --> src/main.rs:30:24
   |
30 |         timer.schedule(now, Some(chrono::Duration::milliseconds(1000)), move || {
   |                        ^^^ expected struct `chrono::datetime::DateTime`, found struct `chrono::DateTime`
   |
   = note: expected type `chrono::datetime::DateTime<_>`
              found type `chrono::DateTime<chrono::Local>`
   = help: here are some functions which might fulfill your needs:
           - .timezone()

error: aborting due to previous error(s)

error: Could not compile `chipclock`.

To learn more, run the command again with --verbose.
```
Please note the 2 different chrono versions ...

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "timer"
-version = "0.1.4"
+version = "0.2.0"
 authors = ["David Rajchenbach-Teller <yoric@rajtel.name>"]
 description = "A simple timer. Use it to schedule execution of closures after a delay or at a given timestamp."
 documentation = "http://yoric.github.io/timer.rs/doc/timer/"
@@ -11,5 +11,5 @@ license = "MPL-2.0"
 
 
 [dependencies]
-chrono = "^0.2"
+chrono = "^0.4"
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -10,12 +10,13 @@ use std::sync::atomic::Ordering as AtomicOrdering;
 use std::sync::{Arc, Mutex, Condvar};
 use std::sync::mpsc::{channel, Sender};
 use std::collections::BinaryHeap;
-use chrono::{Duration, DateTime, UTC};
+use chrono::{Duration, DateTime};
+use chrono::offset::Utc;
 
 /// An item scheduled for delayed execution.
 struct Schedule<T> {
     /// The instant at which to execute.
-    date: DateTime<UTC>,
+    date: DateTime<Utc>,
 
     /// The schedule data.
     data : T,
@@ -173,12 +174,12 @@ impl <T,E> Scheduler<T,E> where E : Executor<T> {
             // If we don't find
             let mut sleep = Sleep::UntilAwakened;
             loop {
-                let now = UTC::now();
+                let now = Utc::now();
                 if let Some(sched) = self.heap.peek() {
                     if sched.date > now {
                         // First item is not ready yet, so we need to
                         // wait until it is or something happens.
-                        sleep = Sleep::AtMost(sched.date - now);
+                        sleep = Sleep::AtMost(sched.date.signed_duration_since(now));
                         break;
                     }
                 } else {
@@ -296,7 +297,7 @@ impl <T> TimerBase<T>
     }
 
     pub fn schedule_with_delay(&self, delay: Duration, data : T) -> Guard {
-        self.schedule_with_date(UTC::now() + delay, data)
+        self.schedule_with_date(Utc::now() + delay, data)
     }
 
     pub fn schedule_with_date<D>(&self, date: DateTime<D>, data : T) -> Guard
@@ -307,7 +308,7 @@ impl <T> TimerBase<T>
 
     pub fn schedule_repeating(&self, repeat: Duration, data : T) -> Guard
     {
-        self.schedule(UTC::now() + repeat, Some(repeat), data)
+        self.schedule(Utc::now() + repeat, Some(repeat), data)
     }
 
     pub fn schedule<D>(&self, date: DateTime<D>, repeat: Option<Duration>, data : T) -> Guard
@@ -315,7 +316,7 @@ impl <T> TimerBase<T>
     {
         let guard = Guard::new();
         self.tx.send(Op::Schedule(Schedule {
-            date: date.with_timezone(&UTC),
+            date: date.with_timezone(&Utc),
             data: data,
             guard: guard.clone(),
             repeat: repeat
@@ -730,7 +731,7 @@ mod tests {
     use std::sync::mpsc::channel;
     use std::sync::{Arc, Mutex};
     use std::thread;
-    use chrono::{Duration, UTC};
+    use chrono::{Duration, Utc};
 
     #[test]
     fn test_schedule_with_delay() {
@@ -741,7 +742,7 @@ mod tests {
         // Schedule a number of callbacks in an arbitrary order, make sure
         // that they are executed in the right order.
         let mut delays = vec![1, 5, 3, -1];
-        let start = UTC::now();
+        let start = Utc::now();
         for i in delays.clone() {
             println!("Scheduling for execution in {} seconds", i);
             let tx = tx.clone();
@@ -753,7 +754,7 @@ mod tests {
 
         delays.sort();
         for (i, msg) in (0..delays.len()).zip(rx.iter()) {
-            let elapsed = (UTC::now() - start).num_seconds();
+            let elapsed = Utc::now().signed_duration_since(start).num_seconds();
             println!("Received message {} after {} seconds", msg, elapsed);
             assert_eq!(msg, delays[i]);
             assert!(delays[i] <= elapsed && elapsed <= delays[i] + 3, "We have waited {} seconds, expecting [{}, {}]", elapsed, delays[i], delays[i] + 3);
@@ -761,7 +762,7 @@ mod tests {
 
         // Now make sure that callbacks that are designed to be executed
         // immediately are executed quickly.
-        let start = UTC::now();
+        let start = Utc::now();
         for i in vec![10, 0] {
             println!("Scheduling for execution in {} seconds", i);
             let tx = tx.clone();
@@ -772,14 +773,14 @@ mod tests {
         }
 
         assert_eq!(rx.recv().unwrap(), 0);
-        assert!(UTC::now() - start <= Duration::seconds(1));
+        assert!(Utc::now().signed_duration_since(start) <= Duration::seconds(1));
     }
 
     #[test]
     fn test_message_timer() {
         let (tx, rx) = channel();
         let timer = MessageTimer::new(tx);
-        let start = UTC::now();
+        let start = Utc::now();
 
         let mut delays = vec!(400, 300, 100, 500, 200);
         for delay in delays.clone() {
@@ -790,7 +791,7 @@ mod tests {
         for delay in delays {
             assert_eq!(rx.recv().unwrap(), delay);
         }
-        assert!(UTC::now() - start <= Duration::seconds(1));
+        assert!(Utc::now().signed_duration_since(start) <= Duration::seconds(1));
     }
 
     #[test]

